### PR TITLE
[Enhancement]: Added the exit function of gitter live chat.

### PR DESCRIPTION
**Pull request by wan230114**
Added the exit function of gitter live chat, press keyboard esc to exit gitter or click anywhere outside the area to exit.

**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [x] Other, please describe: added the exit function of gitter live chat

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [x] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

---

* [ ] DO NOT include files inside `lib` directory.

### CODE REVIEW

**Comment by Koooooo-7** [comment]
I test it,  work correctly.
but I m not sure whether it is necessary.
I think it is better to have a place collect those tips .

File: docs/index.html [hunk 1]
```diff
@@ -106,6 +106,16 @@
   };
   </script>
   <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
+  <script>
+    function closechart(params) {
+      const buts = document.getElementsByClassName("gitter-chat-embed-action-bar-item");
```
Comment (trusktr) [inline docs/index.html:111]: What does `buts` stand for? We should prefer meaningful code that is easy to understand.

File: docs/index.html [hunk 2]
```diff
@@ -106,6 +106,16 @@
...
+      for (let but of buts) {
+        if (but.className == "gitter-chat-embed-action-bar-item gitter-chat-embed-action-bar-item-collapse-chat") { but.click(); }
+      }
+    }
+    document.onkeydown = function () {if (event.keyCode == 27) {closechart()}};
```
Comment (trusktr) [inline docs/index.html:116]: Why is this called `closechart`? Gitter chat is not a chart.

File: docs/index.html [hunk 3]
```diff
@@ -106,6 +106,16 @@
...
+    document.onclick = function (e) {if (e.target.className.indexOf("gitter-open-chat-button") == -1) {closechart()}};
```
Comment (trusktr) [inline docs/index.html:117]: We should not use `.onclick` or any `.on` properties on elements. These are not robust and any other code can ruin these handlers.

We should use `document.addEventListener('click', ...)` and `document.addEventListener('keydown', ...)`.

**Comment by anikethsaha** [comment]
> @anikethsaha What's your thought on `type="module"`? Microsoft has moved onto Chromium-based Edge, so all browsers support it now. In my opinion let's roll with the modern browsers. Do we want to support IE 11? (thinks to himself: please say no, please say no).

I dont think we need that. We want to support IE11. I was actually thinking of finding/searching for breakpoints where its not working properly in IE11 and we should fix it. For CSS we should add some postcss based polyfill . 


> place this code in a separate .js file, and include it via <script> tag. This makes it easier for anyone who wants to use the feature to also include it as a script tag without having to copy/paste the code.

yes, maybe we should move the edit button plugin too in a separate file just for our docs.
I think that should be done in separate PR.

File: docs/index.html (hunk 1 above)
Comment (anikethsaha) [reply docs/index.html:111]: I think it is for button ?

may be it should be rename to `btn` instead or something else, `buts` doesn't sound good 👍

**Review by Koooooo-7 — COMMENTED** [review]
Close by remove Gitter to Discord.

**Pull request closed without merging**